### ISSUE TITLE
Async image upload uses the wrong HTTP status code

### ISSUE THREAD

**Issue by vanillajonathan** [tinymce/tinymce-docs#547]
It checks explicitly for status code 200. But actually any status code from the 2xx series should be okay.

Also the proper thing isn't to return a [200 OK](https://httpstatuses.com/200), it is to return a [201 Created](https://httpstatuses.com/201) which should set the `Location` header pointing to the resource that was created.

> The 201 (Created) status code indicates that the request has been
   fulfilled and has resulted in one or more new resources being
   created.  The primary resource created by the request is identified
   by either a Location header field in the response or, if no Location
   field is received, by the effective request URI.

https://tools.ietf.org/html/rfc7231#section-6.3.2
https://www.tinymce.com/docs/advanced/handle-async-image-uploads/

```js
let location = xhr.getResponseHeader("Location");
success(location);
```

**Comment by fyrkant** [comment]
Thanks for this issue report, a fix for the documentation  and the code in the Uploader in the tinymce core is coming soon!

**Comment by r12s** [comment]
@fyrkant did this get resolved in the core?

**Comment by fyrkant** [comment]
Yes this has been fixed.

**Comment by vanillajonathan** [comment]
Accepting 2xx status code was fixed in c1429783178ac887617a05af1d01bb256a37b44d.

The script still relies on deserializing the message body into a JSON object and reading the "location" property, rather than accepting a HTTP 201 request with the `Location` header set.
